### PR TITLE
[ntuple] fix memory leak in page decompression

### PR DESCRIPTION
**Pull request by jblomer**
plus two minor clang-reported issues.

### CODE REVIEW

File: tree/ntuple/v7/src/RPageStorageFile.cxx [hunk 1]
```diff
@@ -380,11 +380,10 @@ ROOT::Experimental::Detail::RPage ROOT::Experimental::Detail::RPageSourceFile::P
       if (!cachedPage.IsNull())
          return cachedPage;
 
+      pageBuffer = new unsigned char[bytesPacked];
```
Comment (pcanal) [inline tree/ntuple/v7/src/RPageStorageFile.cxx:383]: Could you add a comment on why moving the allocation is solving the memory leak?  (The only obviously deletion is within the "seemingly" unrelated `!element->IsMappable()` and even-though I suspect some calls might lead to adoption, I don't find it/them trivially).

**Review by pcanal — COMMENTED** [review]
See comment.

File: tree/ntuple/v7/src/RPageStorageFile.cxx (hunk 1 above)
Comment (jblomer) [reply tree/ntuple/v7/src/RPageStorageFile.cxx:385]: The problem is in line 381: Before the patch, we could return early having the buffer already allocated (comment added).

Comment (pcanal) [reply tree/ntuple/v7/src/RPageStorageFile.cxx:385]: In addition to this move (not allocating it when not used), would it make sense to update `pageBuffer` to be a unique_ptr?

Comment (jblomer) [reply tree/ntuple/v7/src/RPageStorageFile.cxx]: @pcanal Sorry for leaving this dormant for so long!  It became more attention as the memory leak fixed was also found by clang-tidy.

Changed to a unique pointer, it does help clarifying ownership.

**Review by pcanal — APPROVED** [review]

**Pull request merged**